src/functions-reference/higher-order_functions.Rmd

```diff
@@ -141,6 +142,138 @@ Stan's automatic differentiation.
+Solves the ODE system for the times provided using the Runge Kutta
+Dopri algorithm with the implementation from Boost with additional
+control parameters for the solver.
+
+<!-- vector[]; ode_adams; (function ode, vector initial_state, real initial_time, real[] times, ...); -->
+\index{{\tt \bfseries ode\_adams }!{\tt (function ode, vector initial\_state, real initial\_time, real[] times, ...): vector[]}|hyperpage}
+
+`vector[]` **`ode_adams`**`(function ode, vector initial_state, real initial_time, real[] times, ...)`<br>\newline
+Solves the ODE system for the times provided using the Adams-Moulton
+method with the implementation from CVODES.
+
+<!-- vector[]; ode_adams_tol; (function ode, vector initial_state, real initial_time, real[] times, data real rel_tol, data real abs_tol, data int max_num_steps, ...); -->
+\index{{\tt \bfseries ode\_adams\_tol }!{\tt (function ode, vector initial\_state, real initial\_time, real[] times, data real rel\_tol, data real abs\_tol, data int max\_num\_steps, ...): vector[]}|hyperpage}
+
+`vector[]` **`ode_adams_tol`**`(function ode, vector initial_state, real initial_time, real[] times, data real rel_tol, data real abs_tol, data int max_num_steps, ...)`<br>\newline
+Solves the ODE system for the times provided using the Adams-Moulton
+method with the implementation from CVODES with additional control
+parameters for the CVODES solver.
+
+### Stiff Solver
+
+<!-- vector[]; ode_bdf; (function ode, vector initial_state, real initial_time, real[] times, ...); -->
+\index{{\tt \bfseries ode\_bdf }!{\tt (function ode, vector initial\_state, real initial\_time, real[] times, ...): vector[]}|hyperpage}
+
+`vector[]` **`ode_bdf`**`(function ode, vector initial_state, real initial_time, real[] times, ...)`<br>\newline
+Solves the ODE system for the times provided using the backward
+differentiation formula (BDF) method with the implementation from
+CVODES.
+
+<!-- vector[]; ode_bdf_tol; (function ode, vector initial_state, real initial_time, real[] times, data real rel_tol, data real abs_tol, data int max_num_steps, ...); -->
+\index{{\tt \bfseries ode\_bdf\_tol }!{\tt (function ode, vector initial\_state, real initial\_time, real[] times, data real rel\_tol, data real abs\_tol, data int max\_num\_steps, ...): vector[]}|hyperpage}
+
+`vector[]` **`ode_bdf_tol`**`(function ode, vector initial_state, real initial_time, real[] times, data real rel_tol, data real abs_tol, data int max_num_steps, ...)`<br>\newline
+Solves the ODE system for the times provided using the backward
+differentiation formula (BDF) method with the implementation from
+CVODES with additional control parameters for the CVODES solver.
+
+### Arguments to the ODE solvers
+
+The arguments to the ODE solvers in both the stiff and non-stiff cases
+are as follows.
+
+*   *`ode`*: function literal referring to a function specifying  the
+system of differential equations with signature:
```

Comment:
What does function literal mean? Should we use that same term when talking about the ODE function above?

Reply:
Is this section the same as the one above?

Reply:
> Is this section the same as the one above?

I left a copy of the old integrator docs there. The function signature will be deprecated but I figure it should still have docs so people know how to convert old models. So there is duplicate information.